src/transformers/configuration_utils.py

```diff
@@ -1057,6 +1056,36 @@ def _get_non_default_generation_parameters(self) -> Dict[str, Any]:
 
         return non_default_generation_parameters
 
+    def get_text_config(self, decoder=False) -> "PretrainedConfig":
+        """
+        Returns the config that is meant to be used with text IO. On most models, it is the original config instance
+        itself. On specific composite models, it is under a set of valid names.
+
+        If `decoder` is set to `True`, then only search for decoder config names.
```

Comment:
for my own understanding, what does it mean to search in "decoder config names"? Is it somehow related to a model being an encoder-decoder or decoder-only?

From what I see, `text_encoder` is used in Musicgen only and we never used the `decoder=False` in transformers

Reply:
It is indeed mostly for `musicgen` at the moment. Needing/wanting to use the encoder text config is uncommon, but there is at least one pair of use cases:
1. `resize_token_embeddings` (has `decoder=False`)
2. The tests for `resize_token_embeddings`